### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by kbarnhart**
Some updates to the documentation itself (partially addresses Issue #1108)

My intention in these updates is to focus the majority of the updates associated with @rreinecke's comments (summarized on #1108, originally from [the JOSS review issue](https://github.com/openjournals/joss-reviews/issues/1935)) as improvements to the core documentation. 

A separate PR will update the Readme.md for the GroundwaterDupuitPercolator component and link appropriately to the documentation updated here.

@DavidLitwin @mcflugen could you take a look at this. 

Specifically:
* @mcflugen. Let me know if you think I've described install/update/dependencies well?
* @DavidLitwin Let me know if you think that this (along with the forthcoming PR) is sufficient to addresses @rreinecke's concerns.

### CODE REVIEW

**Comment by DavidLitwin** [comment]
Hi @kbarnhart, It seems that the main concerns were 

1. Finding unit tests
2. Finding locations of other components and their documentation (?)
3. Install instructions and associated dependencies

The documents you have changed seem to me to address these questions assuming that they are themselves findable. I haven't checked all links/pages to make sure that is the case. Do you think it is?

**Comment by kbarnhart** [comment]
@mcflugen does this look OK to you?

File: docs/source/development/practices/continuous_integration.rst [hunk 1]
```diff
@@ -13,10 +13,10 @@ Landlab uses `Travis <https://travis-ci.org>`_ for continuous integration
 testing. The `landlab page on Travis <https://travis-ci.org/landlab/landlab>`_
 shows the latest testing results. A new set of tests are executed whenever
 any changes are pushed to the Landlab repository and with every pull request.
-We currently run test suites for Python versions 2.7, 3.5, and 3.6.
+We currently run test suites for Python versions 3.5, and 3.6.
```
Comment (mcflugen) [inline docs/source/development/practices/continuous_integration.rst]: Should be *3.6*, and *3.7* (and soon to be *3.8*, I hope!).

File: docs/source/development/practices/continuous_integration.rst [hunk 2]
```diff
@@ -13,10 +13,10 @@ Landlab uses `Travis <https://travis-ci.org>`_ for continuous integration
...
 
 Continuous integration for Windows is done on
-`Appveyor <https://ci.appveyor.com>`_ and also tests with Python 2.7, 3.5, and 3.6.
+`Appveyor <https://ci.appveyor.com>`_ and also tests with Python 3.5, and 3.6.
```
Comment (mcflugen) [inline docs/source/development/practices/continuous_integration.rst]: Again, *3.6* and *3.7*.

File: docs/source/install/index.rst [hunk 3]
```diff
@@ -17,6 +17,12 @@ You can install Landlab using either the pip or conda package management tools.
 We distribute Landlab through `PyPI <https://pypi.org/project/landlab/>`_
 and `conda-forge <https://anaconda.org/conda-forge/landlab>`_.
 
+The following instructions will just install Landlab and its dependencies.
+There may be some additional dependencies necessary to run the Landlab
+notebooks or develop a model with Landlab (e.g., spyder). Below we also provide
+instructions to install a conda environment which includes everythin you need
```
Comment (mcflugen) [inline docs/source/install/index.rst]: Either *everythin'* or, maybe, just *everything*. 😄

**Comment by mcflugen** [comment]
I think, to address @rreinecke concern in #1108, we also need to update our main README file, which is quite out of date and not all that useful. I imagine it should just be a TLDR for our longer documentation. That can be done in another pull request though.

**Comment by kbarnhart** [comment]
Thanks for the comments @mcflugen. I've started to address the readme concerns on #1112, in doing so I've realized that the are some other things to change in the main docs. So you'll see another doc PR from me in the next day or so. 

I'm keeping the readme PRs separate from the doc PRs b/c I want to make sure to get the links from the readme into the docs right.

**Comment by kbarnhart** [comment]
[Linking to the JOSS Review Issue](https://github.com/openjournals/joss-reviews/issues/1935)

**Pull request merged**